vis-operators.c

```diff
@@ -181,6 +181,33 @@ static size_t op_cursor(Vis *vis, Text *txt, OperatorContext *c) {
 	return EPOS;
 }
 
+static size_t op_wrap_text(Vis *vis, Text *txt, OperatorContext *c) {
+	size_t pos = text_line_prev(txt, c->range.start), prev_pos;
+
+	do {
+		prev_pos = pos;
+		size_t end = text_line_next(txt, pos);
+		while (pos - prev_pos < vis->textwidth && end > pos)
+			pos = text_longword_end_next(txt, pos);
+		if (pos - prev_pos >= vis->textwidth) {
+			size_t break_pos = text_longword_start_prev(txt, pos);
+			if (break_pos == prev_pos) {
+				/* vis->textwidth is longer than this word, insert line break
+				 * immediately after this word */
+				break_pos = pos + 1;
```

Comment:
Not sure if these direct addition/subtractions to the position are correct. It _looks_ like text_longword_* guarantees that the adjacent character is always whitespace, so this should be OK, but I may have missed something. I'm mostly concerned about jumping into the middle of a unicode character, breaking the encoding.

Might want to use the text iterators here?